pivotreport.php

```diff
@@ -188,3 +188,29 @@ function pivotreport_civicrm_entityTypes(&$entityTypes) {
     'table' => 'civicrm_pivotreport_config',
   ];
 }
+
+/**
+ * Implements hook_civicrm_alterAPIPermissions()
+ */
+function pivotreport_civicrm_alterAPIPermissions($entity, $action, &$params, &$permissions) {
+  switch (true) {
+    case $entity == 'setting' && $action == 'get':
+      $return = $params['return'];
+
+      if (in_array('weekBegins', $return) || in_array('fiscalYearStart', $return)) {
+        $permissions[$entity][$action] = array('access CiviCRM pivot table reports');
+      }
+      break;
+
+    case $entity == 'pivot_report':
+    case $entity == 'pivot_report_config':
+      $permissions[$entity][$action] = array('access CiviCRM pivot table reports');
+      break;
+
+    case $entity == 'optionvalue' && $action == 'get':
+      if ($params['option_group_id'] == 'relative_date_filters') {
+        $permissions[$entity][$action][] = 'access CiviCRM pivot table reports';
```

Comment:
$permissions[$entity][$action] instead of $permissions[$entity][$action][] , right ?

Reply:
`$permissions[$entity][$action] = 'access CiviCRM pivot table reports';` would break it, as `$permissions[$entity][$action]` is supposed to be an array. Since I want to add a new permission, without replacing existing ones, I need to push it into the existing array.